doc/admin/deployment.md

```diff
@@ -75,6 +75,69 @@ for latest information and examples.
 At this point of the process, the servers: and provider: section of the yaml
 file can be left blank and will be populated in the subsequent sections.
 
+#### Auto generate configuration file
+
+DAOS can attempt to produce a server configuration file that makes optimal use
+of hardware on a given set of hosts through the 'dmg config generate' command:
+
+```bash
+$ dmg config generate --help
+ERROR: dmg: Usage:
+  dmg [OPTIONS] config generate [generate-OPTIONS]
+
+Application Options:
+...
+  -l, --host-list=  comma separated list of addresses <ipv4addr/hostname:port>
+...
+
+[generate command options]
+      -p, --num-pmem=  Minimum number of SCM (pmem) devices required per
+                       storage host in DAOS system
+      -n, --num-nvme=  Minimum number of NVMe devices required per storage host
+                       in DAOS system
+      -c, --net-class=[best-available|ethernet|infiniband] Network class
+                       preferred (default: best-available)
+```
+
+The command will output recommended config file if supplied requirements are
+met. Requirements will be derived based on the number of NUMA nodes present on
+the hosts if '--num-pmem' is not specified on the commandline.
+
+- '--num-pmem' specifies the number of persistent memory (pmem) block devices
+that must be present on each host (where the number of pmem devices usually
```

Comment:
The parenthetical explanation is redundant with the explanation of the default below.